Content.Server/Movement/Systems/ContentEyeSystem.cs

```diff
@@ -5,18 +5,25 @@ namespace Content.Server.Movement.Systems;
 
 public sealed class ContentEyeSystem : SharedContentEyeSystem
 {
-    public override void Update(float frameTime)
+    public override void Initialize()
     {
-        base.Update(frameTime);
+        base.Initialize();
+    
+        SubscribeAllEvent<EndOfTargetZoomAnimation>(OnZoomAnimationEnd);
+    }
 
+    private void OnZoomAnimationEnd(EndOfTargetZoomAnimation msg)
+    {
         var query = AllEntityQuery<ContentEyeComponent, SharedEyeComponent>();
 
         while (query.MoveNext(out var uid, out var comp, out var eyeComp))
         {
-            if (eyeComp.Zoom.Equals(comp.TargetZoom))
-                continue;
-
-            UpdateEye(uid, comp, eyeComp, frameTime);
+            if (!eyeComp.Zoom.Equals(comp.TargetZoom))
+            {
+                Sawmill.Debug($"set value {comp.TargetZoom} for zoom animation end");
+                eyeComp.Zoom = comp.TargetZoom;
+                Dirty(eyeComp);
+            }
         }
     }
-}
```

Comment:
This looks very dirty?

Reply:
> This looks very dirty?

Is this question about new code or old code?